third_party/nvfuser/csrc/fusion_segmenter.cpp

```diff
@@ -3145,6 +3172,66 @@ void SegmentedFusion::setCachedHeuristicDataFor(
   heuristic_summary_cache_[group] = std::move(data);
 }
 
+void SegmentedFusion::validate(bool require_disjoint) const {
```

Comment:
Some sanity checks that makes me feel more confident. Had some difficulties in debugging. These checks would have been helpful.